src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -8263,6 +8282,34 @@ public void setIPAddressPreference(String iPAddressPreference) {
     public String getIPAddressPreference() {
         return activeConnectionProperties.getProperty(SQLServerDriverStringProperty.IPADDRESS_PREFERENCE.toString());
     }
+
+
+
+    /** Message handler */
+    private transient ISQLServerMessageHandler serverMessageHandler;
+    
+    /**
+     * Set current message handler
+     * 
+     * @param messageHandler
+     * @return The previously installed message handler (null if none)
+     */
+    @Override
+    public ISQLServerMessageHandler setServerMessageHandler(ISQLServerMessageHandler messageHandler)
+    {
+    	ISQLServerMessageHandler installedMessageHandler = this.serverMessageHandler;
+    	this.serverMessageHandler = messageHandler;
```

Comment:
this overwrites any existing, maybe should check and log a message?

Reply:
Yes a setter usually overwrites previous values...
If you don't want to overwrite it, use the getter and check if you have one...

If you want: I can generate a log message (let me know the LogLevel)   
But from my perspective, I would leave it as is